### PR TITLE
Fix names in plot legends

### PR DESCRIPTION
**Pull request by isaacsas**
Fixes https://github.com/SciML/ModelingToolkit.jl/issues/636

### CODE REVIEW

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by isaacsas** [comment]
@shashi so `nameof.(operation.(states(sys)))` works if the states are terms like `A(t)` but doesn't work if they are syms like `A`? Is there an existing way to handle both?

File: src/utils.jl [hunk 1]
```diff
@@ -159,6 +159,7 @@ tovar(s::Sym) = s
 Base.Symbol(x::Union{Num,Symbolic}) = tosymbol(x)
 tosymbol(x; kwargs...) = x
 tosymbol(x::Sym; kwargs...) = nameof(x)
+tosymbol(x::Term; kwargs...) = nameof(operation(x))
```
Comment (isaacsas) [inline src/utils.jl:162]: @shashi @YingboMa is this reasonable to do, or will it potentially break things elsewhere that rely on the pass through?

**Comment by shashi** [comment]
> Is there an existing way to handle both?

`opname`

**Comment by isaacsas** [comment]
I don't see a merged `opname` function in either ModelingToolkit or SymbolicUtils. Is that something you have in a PR? If you want to give me it I can add it here, or if you merge it I'll rebase and switch to it.

**Comment by isaacsas** [comment]
OK, the old `tosymbol` seems to work for terms if escaping is disabled so I've switched to that. Assuming tests pass this should be good to merge.

**Comment by YingboMa** [comment]
No, this not the right fix. It will totally break when plotting states with derivatives.

**Pull request merged**